Classes/ViewHelpers/Product/IfBestSpecialPriceAvailableViewHelper.php

```diff
@@ -46,34 +42,9 @@ protected static function evaluateCondition($arguments = null)
         return $bestSpecialPrice < $product->getMinPrice();
     }
 
-    /**
-     * Get Frontend User Group
-     *
-     * @return array
-     */
     protected static function getFrontendUserGroupIds(): array
     {
-        $user = $GLOBALS['TSFE']->fe_user->user;
-        if (!$user || !(int)$user['uid']) {
-            return [];
-        }
-
-        $feGroupIds = [];
-
-        $frontendUserRepository = GeneralUtility::makeInstance(
-            FrontendUserRepository::class
-        );
-        $feUser = $frontendUserRepository->findByUid((int)$user['uid']);
-
-        if (!$feUser instanceof FrontendUser) {
-            return [];
-        }
-
-        $feGroups = $feUser->getUsergroup();
-        foreach ($feGroups as $feGroup) {
-            $feGroupIds[] = $feGroup->getUid();
-        }
-
-        return $feGroupIds;
+        $context = GeneralUtility::makeInstance(Context::class);
```

Comment:
Why not using the `__construct` method here? OK, I see the "Make sure to NOT OVERRIDE the constructor." in the doc comment for the `\TYPO3Fluid\Fluid\Core\ViewHelper\AbstractConditionViewHelper` class. But neither this class nor the parent class have an `__construct` method.

I reviewed this on a small mobile device.

Reply:
This is a static function so we cannot use `$this->...` :(